### PR TITLE
Reorganise components

### PR DESCRIPTION
**Pull request by jessie-ross**
**This PR only reorganised components into folders.**
- It does not change code beyond import references.
- It does not rename the files themselves (I plan on doing this as well).

We have ~80 components inside our `/src/components` folder, I think we should reorganise them.

- Aids in auditing our components.
- Improves reuse of components as it is easier to browse through them.
- Allows us to see patterns in our components and better refactor them.
- Starts separating platform code out.
   - Code that isn't tied to Abacus but rather is used as a building block.
   - Makes it easier to assess our platform usage and separate out.
   - Makes it easier for other projects to reuse this code.
   - Is a good code pattern to have.

_Note: It is very easy to reorganise these at a later date so I don't think we need to get too stuck on the specifics._

**Here is the proposed tree:**

```
% tree .   
.
├── Headings.stories.tsx
├── Layout.stories.tsx
├── Layout.test.tsx
├── Layout.tsx
├── RenderErrorBoundary.test.tsx
├── RenderErrorBoundary.tsx
├── RenderErrorView.test.tsx
├── RenderErrorView.tsx
├── __snapshots__
│   └── Layout.test.tsx.snap
├── experiment
│   ├── ExperimentCodeSetup.test.tsx
│   ├── ExperimentCodeSetup.tsx
│   ├── ExperimentStatus.stories.tsx
│   ├── ExperimentStatus.test.tsx
│   ├── ExperimentStatus.tsx
│   ├── __snapshots__
│   │   └── ExperimentCodeSetup.test.tsx.snap
│   ├── experiment-creation
│   │   ├── Audience.test.tsx
│   │   ├── Audience.tsx
│   │   ├── BasicInfo.stories.tsx
│   │   ├── BasicInfo.test.tsx
│   │   ├── BasicInfo.tsx
│   │   ├── Beginning.stories.tsx
│   │   ├── Beginning.test.tsx
│   │   ├── Beginning.tsx
│   │   ├── ExperimentForm.stories.tsx
│   │   ├── ExperimentForm.test.tsx
│   │   ├── ExperimentForm.tsx
│   │   ├── Metrics.test.tsx
│   │   ├── Metrics.tsx
│   │   └── __snapshots__
│   │       ├── Audience.test.tsx.snap
│   │       ├── BasicInfo.test.tsx.snap
│   │       ├── ExperimentForm.test.tsx.snap
│   │       └── Metrics.test.tsx.snap
│   ├── experiment-overview
│   │   ├── AudiencePanel.test.tsx
│   │   ├── AudiencePanel.tsx
│   │   ├── ConclusionsPanel.test.tsx
│   │   ├── ConclusionsPanel.tsx
│   │   ├── ExperimentDetails.test.tsx
│   │   ├── ExperimentDetails.tsx
│   │   ├── GeneralPanel.test.tsx
│   │   ├── GeneralPanel.tsx
│   │   ├── MetricAssignmentsPanel.test.tsx
│   │   ├── MetricAssignmentsPanel.tsx
│   │   ├── SegmentsTable.test.tsx
│   │   ├── SegmentsTable.tsx
│   │   ├── VariationsTable.tsx
│   │   └── __snapshots__
│   │       ├── AudiencePanel.test.tsx.snap
│   │       └── ExperimentDetails.test.tsx.snap
│   ├── experiment-results
│   │   ├── CondensedLatestAnalyses.tsx
│   │   ├── ExperimentDebug.test.tsx
│   │   ├── ExperimentDebug.tsx
│   │   ├── ExperimentResults.stories.tsx
│   │   ├── ExperimentResults.test.tsx
│   │   ├── ExperimentResults.tsx
│   │   ├── FullLatestAnalyses.tsx
│   │   ├── ParticipantCounts.tsx
│   │   ├── RecommendationString.tsx
│   │   └── __snapshots__
│   │       ├── ExperimentDebug.test.tsx.snap
│   │       └── ExperimentResults.test.tsx.snap
│   ├── experiment-table
│   │   ├── ExperimentsTable.stories.tsx
│   │   ├── ExperimentsTable.test.tsx
│   │   └── ExperimentsTable.tsx
│   └── experiment-view-frame
│       ├── ExperimentDisableButton.test.tsx
│       ├── ExperimentDisableButton.tsx
│       ├── ExperimentPageView.test.tsx
│       ├── ExperimentPageView.tsx
│       ├── ExperimentRunButton.test.tsx
│       ├── ExperimentRunButton.tsx
│       └── __snapshots__
│           ├── ExperimentDisableButton.test.tsx.snap
│           └── ExperimentRunButton.test.tsx.snap
├── metrics
│   ├── MetricFormFields.test.tsx
│   ├── MetricFormFields.tsx
│   ├── MetricsTable.stories.tsx
│   ├── MetricsTable.test.tsx
│   ├── MetricsTable.tsx
│   └── __snapshots__
│       └── MetricFormFields.test.tsx.snap
├── platform-general
│   ├── Autocomplete.test.tsx
│   ├── Autocomplete.tsx
│   ├── Code.tsx
│   ├── DatetimeText.test.tsx
│   ├── DatetimeText.tsx
│   ├── DebugOutput.tsx
│   ├── LabelValueTable.stories.tsx
│   ├── LabelValueTable.test.tsx
│   ├── LabelValueTable.tsx
│   ├── LoadingButtonContainer.tsx
│   ├── MoreMenu.tsx
│   └── __snapshots__
│       └── Autocomplete.test.tsx.snap
└── tags
    ├── TagFormFields.test.tsx
    ├── TagFormFields.tsx
    ├── TagsTable.test.tsx
    ├── TagsTable.tsx
    └── __snapshots__
        └── TagFormFields.test.tsx.snap

18 directories, 92 files
```

**And the current tree:**

```
% tree .
.
├── AudiencePanel.test.tsx
├── AudiencePanel.tsx
├── Autocomplete.test.tsx
├── Autocomplete.tsx
├── Code.tsx
├── ConclusionsPanel.test.tsx
├── ConclusionsPanel.tsx
├── DatetimeText.test.tsx
├── DatetimeText.tsx
├── DebugOutput.tsx
├── ExperimentCodeSetup.test.tsx
├── ExperimentCodeSetup.tsx
├── ExperimentDetails.test.tsx
├── ExperimentDetails.tsx
├── ExperimentDisableButton.test.tsx
├── ExperimentDisableButton.tsx
├── ExperimentPageView.test.tsx
├── ExperimentPageView.tsx
├── ExperimentRunButton.test.tsx
├── ExperimentRunButton.tsx
├── ExperimentStatus.stories.tsx
├── ExperimentStatus.test.tsx
├── ExperimentStatus.tsx
├── ExperimentsTable.stories.tsx
├── ExperimentsTable.test.tsx
├── ExperimentsTable.tsx
├── GeneralPanel.test.tsx
├── GeneralPanel.tsx
├── Headings.stories.tsx
├── LabelValueTable.stories.tsx
├── LabelValueTable.test.tsx
├── LabelValueTable.tsx
├── Layout.stories.tsx
├── Layout.test.tsx
├── Layout.tsx
├── LoadingButtonContainer.tsx
├── MetricAssignmentsPanel.test.tsx
├── MetricAssignmentsPanel.tsx
├── MetricFormFields.test.tsx
├── MetricFormFields.tsx
├── MetricsTable.stories.tsx
├── MetricsTable.test.tsx
├── MetricsTable.tsx
├── MoreMenu.tsx
├── RenderErrorBoundary.test.tsx
├── RenderErrorBoundary.tsx
├── RenderErrorView.test.tsx
├── RenderErrorView.tsx
├── SegmentsTable.test.tsx
├── SegmentsTable.tsx
├── TagFormFields.test.tsx
├── TagFormFields.tsx
├── TagsTable.test.tsx
├── TagsTable.tsx
├── VariationsTable.tsx
├── __snapshots__
│   ├── AudiencePanel.test.tsx.snap
│   ├── Autocomplete.test.tsx.snap
│   ├── ExperimentCodeSetup.test.tsx.snap
│   ├── ExperimentDetails.test.tsx.snap
│   ├── ExperimentDisableButton.test.tsx.snap
│   ├── ExperimentRunButton.test.tsx.snap
│   ├── Layout.test.tsx.snap
│   ├── MetricFormFields.test.tsx.snap
│   └── TagFormFields.test.tsx.snap
├── experiment-creation
│   ├── Audience.test.tsx
│   ├── Audience.tsx
│   ├── BasicInfo.stories.tsx
│   ├── BasicInfo.test.tsx
│   ├── BasicInfo.tsx
│   ├── Beginning.stories.tsx
│   ├── Beginning.test.tsx
│   ├── Beginning.tsx
│   ├── ExperimentForm.stories.tsx
│   ├── ExperimentForm.test.tsx
│   ├── ExperimentForm.tsx
│   ├── Metrics.test.tsx
│   ├── Metrics.tsx
│   └── __snapshots__
│       ├── Audience.test.tsx.snap
│       ├── BasicInfo.test.tsx.snap
│       ├── ExperimentForm.test.tsx.snap
│       └── Metrics.test.tsx.snap
└── experiment-results
    ├── CondensedLatestAnalyses.tsx
    ├── ExperimentDebug.test.tsx
    ├── ExperimentDebug.tsx
    ├── ExperimentResults.stories.tsx
    ├── ExperimentResults.test.tsx
    ├── ExperimentResults.tsx
    ├── FullLatestAnalyses.tsx
    ├── ParticipantCounts.tsx
    ├── RecommendationString.tsx
    └── __snapshots__
        ├── ExperimentDebug.test.tsx.snap
        └── ExperimentResults.test.tsx.snap

5 directories, 92 files
```

## How has this been tested?

- Automated tests that cover added/changed functionality
- Manual testing with production data (locally)
- Manual testing with mock data (locally or on Vercel)
- Additional manual testing instructions (please specify):

### CODE REVIEW

**Comment by yanirs** [comment]
I agree it's time to reorganise the component directory, but I think the proposed structured should be tweaked a bit:

* Change `experiment/` to `experiments/` (consistency with `metrics/` and `tags/`)
* Change `experiment/experiment-creation` to `experiments/creation` (no need to repeat the word `experiment` in nested directories)
* Change `experiment/experiment-overview` to `experiments/overview` (same)
* Change `experiment/experiment-results` to `experiments/results` (same)
* Move the files in `experiment/experiment-table` to `experiments/` (consistency with `metrics/` and `tags/`)
* Change `experiment/experiment-view-frame` to `experiments/navigation` or `experiments/top-bar` (frame is a bit confusing)
* Maybe rename `platform-general` to `general` or `util`? `platform` has a different meaning in the context of ExPlat.

What do you think?

**Comment by jessie-ross** [comment]
Good points there!

> - Move the files in experiment/experiment-table to experiments/ (consistency with metrics/ and tags/)
> - Change experiment/experiment-view-frame to experiments/navigation or experiments/top-bar (frame is a bit confusing)

Hmm, how about using `single-view` (nesting `overview` and `results`) and `multi-view` 

What about edit...

> Maybe rename platform-general to general or util? platform has a different meaning in the context of ExPlat.

Ah of course, that is a pity. We could also use something like `foundational` but perhaps `general` is simple enough for now, we can always revisit later.

**Comment by yanirs** [comment]
> > * Move the files in experiment/experiment-table to experiments/ (consistency with metrics/ and tags/)
> > * Change experiment/experiment-view-frame to experiments/navigation or experiments/top-bar (frame is a bit confusing)
> 
> Hmm, how about using `single-view` (nesting `overview` and `results`) and `multi-view` for the table.
> 
> I think we can depart from `metrics` and `tags` as they don't have single views yet.

Since this is already nested under `src/components/experiments`, another level of nesting feels unnecessary/excessive to me, but if you feel strongly about adding `single-view` and `multi-view`, I don't mind. :slightly_smiling_face: `single-view` would be a good home for `ExperimentCodeSetup`.

> > Maybe rename platform-general to general or util? platform has a different meaning in the context of ExPlat.
> 
> Ah of course, that is a pity. We could also use something like `foundational` but perhaps `general` is simple enough for now, we can always revisit later.

Yeah, `general` is good enough for now. Can also go with `generic` or `reusable`, but none of these names is amazing.

**Comment by jessie-ross** [comment]
## Update

- Renamed `platform-general` to just `general`
- Used `single-view` and `multi-view`
- Additionally renamed `experiment-creation` to `wizard`

Here is the new tree:

```
% tree .
.
├── Headings.stories.tsx
├── Layout.stories.tsx
├── Layout.test.tsx
├── Layout.tsx
├── RenderErrorBoundary.test.tsx
├── RenderErrorBoundary.tsx
├── RenderErrorView.test.tsx
├── RenderErrorView.tsx
├── __snapshots__
│   └── Layout.test.tsx.snap
├── experiment
│   ├── ExperimentCodeSetup.test.tsx
│   ├── ExperimentCodeSetup.tsx
│   ├── ExperimentStatus.stories.tsx
│   ├── ExperimentStatus.test.tsx
│   ├── ExperimentStatus.tsx
│   ├── __snapshots__
│   │   └── ExperimentCodeSetup.test.tsx.snap
│   ├── multi-view
│   │   ├── ExperimentsTable.stories.tsx
│   │   ├── ExperimentsTable.test.tsx
│   │   └── ExperimentsTable.tsx
│   ├── single-view
│   │   ├── ExperimentDisableButton.test.tsx
│   │   ├── ExperimentDisableButton.tsx
│   │   ├── ExperimentPageView.test.tsx
│   │   ├── ExperimentPageView.tsx
│   │   ├── ExperimentRunButton.test.tsx
│   │   ├── ExperimentRunButton.tsx
│   │   ├── __snapshots__
│   │   │   ├── ExperimentDisableButton.test.tsx.snap
│   │   │   └── ExperimentRunButton.test.tsx.snap
│   │   ├── overview
│   │   │   ├── AudiencePanel.test.tsx
│   │   │   ├── AudiencePanel.tsx
│   │   │   ├── ConclusionsPanel.test.tsx
│   │   │   ├── ConclusionsPanel.tsx
│   │   │   ├── ExperimentDetails.test.tsx
│   │   │   ├── ExperimentDetails.tsx
│   │   │   ├── GeneralPanel.test.tsx
│   │   │   ├── GeneralPanel.tsx
│   │   │   ├── MetricAssignmentsPanel.test.tsx
│   │   │   ├── MetricAssignmentsPanel.tsx
│   │   │   ├── SegmentsTable.test.tsx
│   │   │   ├── SegmentsTable.tsx
│   │   │   ├── VariationsTable.tsx
│   │   │   └── __snapshots__
│   │   │       ├── AudiencePanel.test.tsx.snap
│   │   │       └── ExperimentDetails.test.tsx.snap
│   │   └── results
│   │       ├── CondensedLatestAnalyses.tsx
│   │       ├── ExperimentDebug.test.tsx
│   │       ├── ExperimentDebug.tsx
│   │       ├── ExperimentResults.stories.tsx
│   │       ├── ExperimentResults.test.tsx
│   │       ├── ExperimentResults.tsx
│   │       ├── FullLatestAnalyses.tsx
│   │       ├── ParticipantCounts.tsx
│   │       ├── RecommendationString.tsx
│   │       └── __snapshots__
│   │           ├── ExperimentDebug.test.tsx.snap
│   │           └── ExperimentResults.test.tsx.snap
│   └── wizard
│       ├── Audience.test.tsx
│       ├── Audience.tsx
│       ├── BasicInfo.stories.tsx
│       ├── BasicInfo.test.tsx
│       ├── BasicInfo.tsx
│       ├── Beginning.stories.tsx
│       ├── Beginning.test.tsx
│       ├── Beginning.tsx
│       ├── ExperimentForm.stories.tsx
│       ├── ExperimentForm.test.tsx
│       ├── ExperimentForm.tsx
│       ├── Metrics.test.tsx
│       ├── Metrics.tsx
│       └── __snapshots__
│           ├── Audience.test.tsx.snap
│           ├── BasicInfo.test.tsx.snap
│           ├── ExperimentForm.test.tsx.snap
│           └── Metrics.test.tsx.snap
├── general
│   ├── Autocomplete.test.tsx
│   ├── Autocomplete.tsx
│   ├── Code.tsx
│   ├── DatetimeText.test.tsx
│   ├── DatetimeText.tsx
│   ├── DebugOutput.tsx
│   ├── LabelValueTable.stories.tsx
│   ├── LabelValueTable.test.tsx
│   ├── LabelValueTable.tsx
│   ├── LoadingButtonContainer.tsx
│   ├── MoreMenu.tsx
│   └── __snapshots__
│       └── Autocomplete.test.tsx.snap
├── metrics
│   ├── MetricFormFields.test.tsx
│   ├── MetricFormFields.tsx
│   ├── MetricsTable.stories.tsx
│   ├── MetricsTable.test.tsx
│   ├── MetricsTable.tsx
│   └── __snapshots__
│       └── MetricFormFields.test.tsx.snap
└── tags
    ├── TagFormFields.test.tsx
    ├── TagFormFields.tsx
    ├── TagsTable.test.tsx
    ├── TagsTable.tsx
    └── __snapshots__
        └── TagFormFields.test.tsx.snap
```

After this PR I will be slowly correcting file names as I come across them.

**Comment by yanirs** [comment]
Thanks! The only thing that's missing from the above list is:

> * Change `experiment/` to `experiments/` (consistency with `metrics/` and `tags/`)

Did you miss it or is there a reason for the inconsistency?

Also, should we move the remaining components (`Layout` and `RenderError*`) from the root directory to `general/`? I think we can delete `Headings.stories.tsx`, as it's not particularly useful.

**Comment by jessie-ross** [comment]
> Did you miss it or is there a reason for the inconsistency?

Oops I missed that.


> I think we can delete Headings.stories.tsx, as it's not particularly useful.

Why not 😃 

> Also, should we move the remaining components (Layout and RenderError*) from the root directory to general/? 

Yeah I was trying not to do that, that was why I named it "platform-general" before. I meant `general` to be for self-contained components not tied to the rest of the app, which we should be keeping self-contained for simplicity - so it helps to keep them seperate.

Maybe we could put Layout and Co into `page-parts`, `page-constituents`, or `page-infra`? 

---
In the future I'd like to split Layout into three parts:
- Header
- Footer
- The part that updates the title

And then we just put header and footer into `/App.tsx` before and after the router.

"The part that updates the title" can just be a hook that is used in `/pages`(e.g. `usePageTitle('Experiments')`), and then we don't have the strange pattern of the header and footer being rendered within the page route.

**Comment by yanirs** [comment]
> Maybe we could put Layout and Co into `page-parts`, `page-constituents`, or `page-infra`?

Let's go with `page-parts` :slightly_smiling_face:

File: src/components/experiments/single-view/ExperimentPageView.tsx [hunk 1]
```diff
@@ -19,17 +19,17 @@ import ExperimentsApi from 'src/api/ExperimentsApi'
...
+import ExperimentDisableButton from 'src/components/experiments/single-view/ExperimentDisableButton'
+import ExperimentDetails from 'src/components/experiments/single-view/overview/ExperimentDetails'
+import Layout from 'src/components/page-parts/Layout'
 import { Analysis, ExperimentFull, Status } from 'src/lib/schemas'
 import { useDataLoadingError, useDataSource } from 'src/utils/data-loading'
 import { createUnresolvingPromise, or } from 'src/utils/general'
 
-import ExperimentDebug from './experiment-results/ExperimentDebug'
-import ExperimentResults from './experiment-results/ExperimentResults'
 import ExperimentRunButton from './ExperimentRunButton'
```
Comment (yanirs) [inline src/components/experiments/single-view/ExperimentPageView.tsx:30]: It's a bit weird that some component imports are relative and some are absolute, especially in cases like this one when comparing `ExperimentRunButton` and `ExperimentDisableButton`. We can take care of it in a separate PR, though.

**Review by yanirs — APPROVED** [review]
As noted inline, it'd be nice to use absolute and relative imports consistently, but it's not a priority. :ship:

**Pull request merged**